resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/server/servlet/HttpServletResponseHeaders.java

```diff
@@ -150,13 +150,32 @@ public void putAll(Map<? extends String, ? extends List<Object>> map)
       for (Map.Entry<? extends String, ? extends List<Object>> entry : map.entrySet())
       {
          List<Object> objs = entry.getValue();
-         for (Object obj : objs)
-         {
-            add(entry.getKey(), obj);
+         // When the header key does not exist, undertow creates it and saves this value.
+         // When the header key exists undertow clears the existing values and saves this value.
+         // All subsequent values must be added so they will be appended properly.
+         putResponseHeader(entry.getKey(), objs.get(0)); // rls test
```

Comment:
This is going to fail in case we get an empty list (yes, an edge case, but still... a check on the size would probably be safe). Or perhaps an empty list has a special meaning?